// myModel/access_api.js
var request = require('request');

/*
 * Model access to api server 
 */

var base_url = 'http://49.212.141.115:3000';

exports.regist = function(title, contents){
    var options = {
        url  : base_url + "/regist/yarimizu", 
        body : "title=" + title + "&contents=" + contents,
        json : true 
    };
    request.post(options, function(err, response, body) {
        console.log(response.statusCode);
        if(!err && response.statusCode === 200) {
            console.log("regist : return from api server");
            console.log(body);
        };
    });
}

exports.getAll = function(){
    var options = {
        url  : base_url + "/all/yarimizu", 
        json : true 
    };
    request.get(options, function(err, response, body) {
        console.log(response.statusCode);
        if(!err && response.statusCode === 200) {
            console.log("get All : return from api server");
            console.log(body);
            exports.getAllCallback(body);
        };
    });
};

exports.getAllCallback = function() {
    console.log("this is getAll callback");
};

exports.getSingle = function(title){
    var options = {
        url  : base_url + "/title/yarimizu?title=" + title, 
        json : true 
    };
    request.get(options, function(err, response, body) {
        if(!err && response.statusCode === 200) {
            console.log("get single : return from api server");
            console.log(body);
            exports.getSingleCallback(body);
        };
    });
};

exports.getSingleCallback = function() {
    console.log("this is getSingle callback");
};
